Deploy metadata in end to end tests

welansari opened the pull request:
- Deploy metadata from chart
- Wait for metadata components to start in order : repd -> bucketd -> cloudserver
- Manually add the proper endpoint in Cloudserver config
- Only deploy metadata in `end2end-sharded`

Issue: ZENKO-4414

Commit:
deploy metadata in end2end sharded tests

Issue: ZENKO-4414

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/zenko/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

Commit:
ensure all metadata components are started properly

Metadata components need to be started in the order
repd -> bucketd -> cloudserver
as they are not able to recover on their own when a
component they depend on is not available at startup.

Issue: ZENKO-4414

Commit:
remove retry count from test ring bucket name

Configuration of the test environement is done through an image built in the build-test-image job,
this image contains the list of location details used to initialize the locations. This list isn't
updated when only retrying the end to end test jobs, and the bucket is create using the env var passed
to the pod. So we endup creating the bucket with the updated retry count but configure the location
with the old bucket name which fails as pensieve won't be able to find that bucket.

Issue: ZENKO-4414

Commit:
only deploy metadata when needed

Issue: ZENKO-4414

francoisferrand commented on .github/actions/deploy/action.yaml:
```diff
@@ -83,6 +83,11 @@ runs:
       working-directory: ./.github/scripts/end2end/operator
+    - name: Deploy metadata
+      shell: bash
+      run: bash deploy-metadata.sh
```
We are already running with bash, so make it executable and execute just:

```suggestion
      run: ./deploy-metadata.sh
```

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4414 contains:


* `2.10.12`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.13`


Please check the `Fix Version/s` of ZENKO-4414, or the target
branch of this pull request.

francoisferrand commented on .github/actions/deploy/action.yaml:
```diff
@@ -83,6 +83,11 @@ runs:
+      shell: bash
+      run: bash deploy-metadata.sh
+      working-directory: ./.github/scripts/end2end
+      if: ${{ env.ENABLE_RING_TESTS == 'true' }}
```
better to have it as an `input` rather than an env variable
(the input parameter may however takes its "default" value from env)

francoisferrand commented on .github/scripts/end2end/deploy-metadata.sh:
```diff
@@ -0,0 +1,47 @@
+# wait for all bucketd pods to start serving port 9000
+wait_for_all_pods_behind_services metadata-bucketd metadata 9000 60
+
+# manually add "s3c.local" to the rest endpoints list as it's not configurable in the chart
```
why do we need `s3c.local` ?

i did not need such patch on my artesca cluster (in namespace `default` i think), just installed the chart and created the artesca location pointed at `s3c-cloudserver.default.svc.cluster.local`

welansari replied:
Cloudserver responds with an error when the host used is not declared in its config file. I've tried with the k8s service endpoint and it doesn't work.

francoisferrand replied:
maybe this comes from the namespace ("metadata") you deploy to, which may not be taken into account?
in my tests this is using the default namespace, with no issue...

welansari replied:
No even with the default namespace i got the same behaviour, which to me, is the expected behaviour. The normal cloudserver we deploy in Zenko has all the k8s endpoints in its config.

francoisferrand commented on .github/workflows/end2end.yaml:
```diff
@@ -556,8 +553,9 @@ jobs:
-      # - scality-cloud
+    env:
+      ENABLE_RING_TESTS: "true"
+      GIT_ACCESS_TOKEN: ${{ secrets.GIT_ACCESS_TOKEN }}
```
credentials should not be added to env, esp. at such broad "scale" : best to pass them via inputs, in the place where they are needed

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue ZENKO-4414 contains:


* `2.10.12`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `2.10.13`

* `2.11.0`


Please check the `Fix Version/s` of ZENKO-4414, or the target
branch of this pull request.

williamlardier commented on .github/scripts/end2end/deploy-metadata.sh:
```diff
@@ -0,0 +1,47 @@
+# clone the metadata repository
+git init metadata
+cd metadata
+git fetch --depth 1 --no-tags https://${GIT_ACCESS_TOKEN}@github.com/scality/metadata.git
+git checkout FETCH_HEAD
```
Do we want to support a specific metadata version? What if the latest commit of the default branch introduced a breaking change or a bug/limitation to be fixed in another PR before a release is made?

francoisferrand replied:
probably not worth it:
* this chart is mostly not maintained
* it references "unknown" version of metadata
* ...consequently this is not a long term solution, just a stop-gap measure until we switch to using metadata-operator to deploy a "recent" metadata : but at least it allows us to test OOB already...

williamlardier replied:
No problem, I'm fine if that's not that maintained

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

welansari commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.11`



The following branches will **NOT** be impacted:

* `development/2.10`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# Queue build failed




The corresponding build for the queue failed:

- Checkout the [status page](https://bert-e.scality.net/scality/zenko).
- Identify the failing build and review the logs.
- If no issue is found, re-run the build.
- If an issue is identified, checkout the steps below to remove
  the pull request from the queue for further analysis and maybe rebase/merge.

<details>
  <summary><b>Remove the pull request from the queue</b></summary>

- Add a `/wait` comment on this pull request.
- Click on login on the [status page](https://bert-e.scality.net/scality/zenko).
- Go into the [manage](https://bert-e.scality.net/scality/zenko/manage) page.
- Find the option called `Rebuild the queue` and click on it.
  Bert-E will loop again on all pull requests to put the valid ones
  in the queue again, while skipping the one with the `/wait` comment.
- Wait for the new queue to merge, then merge/rebase your pull request
  with the latest changes to then work on a proper fix.
- Once the issue is fixed, delete the `/wait` comment and
  follow the usual process to merge the pull request.

</details>

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.11`



The following branches have **NOT** changed:

* `development/2.10`
* `development/2.5`
* `development/2.6`
* `development/2.7`
* `development/2.8`
* `development/2.9`



Please check the status of the associated issue ZENKO-4414.

Goodbye kerkesni.
